tests/language-feature/ifunc/ifunc.slang

```diff
@@ -0,0 +1,40 @@
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-shaderobj -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-dx12 -use-dxil -shaderobj -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-cpu -shaderobj -output-using-type
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-vk -shaderobj -output-using-type
+
+struct Functor : IFunc<int, int, bool>
+{
+    int __call(int p, bool t)
```

Comment:
I am wondering if we can have `out` or `inout` modifiers on the parameters of `__call`

Reply:
That is another restriction of variadic generics, we can't current have them.
It is easy to support `inout` on the whole parameter pack so that all parameters are `inout`, but more difficult to allow each individual parameters to have different directions, because we don't encode `inout` as a type of parameter, but rather a modifier on the parameter itself. Since `inout` is not a part of the type, it cannot be expressed in the `IFunc<...>` interface.